api/commands/build.ts

```diff
@@ -56,52 +55,77 @@ export class Builder {
       this.compiledGraph.operations.map(o => this.buildOperation(o)),
       this.compiledGraph.assertions.map(a => this.buildAssertion(a))
     );
-
+    const allActionNames = allActions.map(n => n.name);
     const actionNameMap: { [name: string]: dataform.IExecutionAction } = {};
-
     allActions.forEach(action => (actionNameMap[action.name] = action));
 
-    // Include only tagged actions
-    const taggedActions =
+    // Actions selected with --tag option should be included
+    const tagFilteredTables =
```

Comment:
suggest defining a function so that we can stop repeating this block:

```
const hasAnyTag = (action: { tags: string[] }) => !this.runConfig.tags || this.runConfig.tags.length === 0 || action.tags.some(actionTag => !!this.runConfig.tags.find(actionTag));
```

then do something like:

```
const allTaggedActions: dataform.IExecutionAction[] = [].concat(
  this.compiledGraph.tables.filter(table => hasAnyTag(table)),
  this.compiledGraph.operations.filter(table => hasAnyTag(table)),
  this.compiledGraph.assertions.filter(table => hasAnyTag(table)),
);
```

Reply:
Done